### PR TITLE
Added weighted moving average to the charts

### PR DESCRIPTION
**Pull request by tlder**

### CODE REVIEW

**Comment by nahuef** [comment]
Hey! Thanks for the PR, I'll check it out on Tuesday!!

File: charts.go [hunk 1]
```diff
@@ -41,9 +41,22 @@ func ScenarioLineChart(scen *Scenario) template.HTML {
...
+				Name:  fmt.Sprintf("%v: %v. Grouped: %v", SimplifyDate(date), dateWMA.Avg, dateWMA.Grouped),
+				Value: dateWMA.Avg,
+			})
+		}
+	}
+
 	line.SetXAxis(maxDates).
 		AddSeries("Max scores", maxScores).
 		AddSeries("Average scores", avgScores).
+		AddSeries(strconv.Itoa(DefaultWMAWindow)+"-day Moving Average", wmaScores).
```
Comment (nahuef) [inline charts.go]: image

Casing should match to be consistent with the rest:
`-day moving average`

File: stats.go [hunk 2]
```diff
@@ -100,6 +116,17 @@ func scenarioWorker(scen *Scenario, sortedTimesPlayed *[]*Scenario, uniqueDays *
 	scen.ChartByDate = ScenarioLineChart(scen)
 }
 
+func calculateWMA(scen *Scenario) {
+	wma := NewWMA(DefaultWMAWindow)
+	for _, dateScores := range scen.ByDateScores {
+		for date, scores := range dateScores {
+			wma.Add(scores...)
+			avg, count := wma.Average()
+			scen.ByDateWMA = append(scen.ByDateWMA, map[string]DateWMA{date: {Avg: float64(int(avg*10)) / 10, Grouped: count}})
```
Comment (nahuef) [inline stats.go]: I feel like this line could be shorter and do less.
How would you feel about moving the following to a separate line?
`DateWMA{date: {Avg: float64(int(avg*10)) / 10, Grouped: count}}`

Resulting in:
```
dateWMA := DateWMA{Avg: float64(int(avg*10)) / 10, Grouped: count}
scen.ByDateWMA = append(scen.ByDateWMA, map[string]DateWMA{date: dateWMA})
```

**Review by nahuef — CHANGES_REQUESTED** [review]
Good PR, thank you for contributing!!! 
Just a few comments.

- The word `weight` could be changed for `score` right? I feel like it would be easier to understand if we do that.
- We should update the Features section on the main README to include this one, at this line:
`Toggle max scores and average scores lines by clicking on the legend at the top of the chart.`

**Comment by tlder** [comment]
Thanks for reviewing my PR. 

I haven't coded anything in Go before so it's a learning experience.

>     * The word `weight` could be changed for `score` right? I feel like it would be easier to understand if we do that.

In which file? `wma.go`?

My intent was to give scores achieved on the same day the same 'weight' when calculating the weighted moving average, with weight equal to `N` on the most recent day and decaying by 1 each previous day within the `N`-day window.

File: stats.go (hunk 2 above)
Comment (tlder) [reply stats.go]: I've made this change but my PR has broken the unit test in stats_test.go. I'll fix that shortly.

**Comment by nahuef** [comment]
> Thanks for reviewing my PR.
> 
> I haven't coded anything in Go before so it's a learning experience.
> 
> > ```
> > * The word `weight` could be changed for `score` right? I feel like it would be easier to understand if we do that.
> > ```
> 
> In which file? `wma.go`?
> 
> My intent was to give scores achieved on the same day the same 'weight' when calculating the weighted moving average, with weight equal to `N` on the most recent day and decaying by 1 each previous day within the `N`-day window.

Oh ok, my bad!! Sounds good!

Then I think we should explain that feature/how it works in the README, maybe in a line below the one you updated, what do you think?

File: stats.go (hunk 2 above)
Comment (nahuef) [reply stats.go]: makes sense, thank you!

**Comment by tlder** [comment]
I've detailed the moving average in the README. Let me know if it's ok. Thanks.

**Comment by nahuef** [comment]
Merged! Thank you for your contribution!!

**Pull request merged**